Core/Utilities/Pool.py

```diff
@@ -0,0 +1,61 @@
+"""
+An abstract class that provides methods to execute calls asynchronously in a certain Pool (ProcessPool or ThreadPool). 
+It should not be used directly, you should use in a subclass.
+"""
+
+from abc import ABCMeta, abstractmethod
+
+__RCSID__ = "$Id$"
+
+class Pool( object ): #TODO: python3 class Pool(metaclass=ABCMeta):
+  __metaclass__ = ABCMeta
+  
+  @abstractmethod
+  def getMaxThreads( self ):
+    """
+    It returns the maximum threads.
+    """
+    pass
+  
+  @abstractmethod
+  def getMinThreads( self ):
+    """
+    It returns the minimum threads
+    """
+    pass
+  
+  @abstractmethod
+  def numWorkingThreads( self ):
+    """
+    It returns the number of running threads
+    """
+    pass
+  
+  @abstractmethod
+  def numWaitingThreads( self ):
+    """
+    It returns the number of waiting threads
+    """
+    pass
+  
+  @abstractmethod
+  def generateJobAndQueueIt( self,
+                             oCallable,
+                             args = None,
+                             kwargs = None,
+                             sTJId = None,
+                             oCallback = None,
+                             oExceptionCallback = None,
+                             blocking = True ):
+    """
+    :param object oCallable the method which will be executed on a thread.
```

Comment:
there need to be colons after the parameter names, e.g.
`:param object oCallable: the method...`
also after return, i.e.:
`:return:`